framework/src/modules/interoperability/sidechain/commands/cc_update.ts

```diff
@@ -12,18 +12,421 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-import { CommandExecuteContext } from '../../../../node/state_machine';
-import { BaseCommand } from '../../../base_command';
-import { COMMAND_ID_SIDECHAIN_CCU } from '../../constants';
-import { crossChainUpdateTransactionParams } from '../../schema';
+import { codec } from '@liskhq/lisk-codec';
+import { hash, verifyWeightedAggSig } from '@liskhq/lisk-cryptography';
+import { regularMerkleTree, sparseMerkleTree } from '@liskhq/lisk-tree';
+import { LiskValidationError, validator } from '@liskhq/lisk-validator';
+import { certificateSchema } from '../../../../node/consensus/certificate_generation/schema';
+import { Certificate } from '../../../../node/consensus/certificate_generation/types';
+import {
+	CommandExecuteContext,
+	CommandVerifyContext,
+	VerificationResult,
+	VerifyStatus,
+} from '../../../../node/state_machine';
+import { createBeforeSendCCMsgAPIContext } from '../../../../testing';
+import { BaseInteroperabilityCommand } from '../../base_interoperability_command';
+import {
+	CHAIN_ACTIVE,
+	CHAIN_REGISTERED,
+	CHAIN_TERMINATED,
+	COMMAND_ID_SIDECHAIN_CCU,
+	CROSS_CHAIN_COMMAND_ID_REGISTRATION,
+	LIVENESS_LIMIT,
+	MESSAGE_TAG_CERTIFICATE,
+	SMT_KEY_LENGTH,
+	STORE_PREFIX_CHAIN_DATA,
+	STORE_PREFIX_CHANNEL_DATA,
+	STORE_PREFIX_OUTBOX_ROOT,
+	VALID_BLS_KEY_LENGTH,
+} from '../../constants';
+import { createCCMsgBeforeSendContext } from '../../context';
+import {
+	ccmSchema,
+	chainAccountSchema,
+	channelSchema,
+	crossChainUpdateTransactionParams,
+} from '../../schema';
+import {
+	ActiveValidators,
+	BFTAPI,
+	CCMsg,
+	ChainAccount,
+	ChannelData,
+	CrossChainCommandDependencies,
+	CrossChainUpdateTransactionParams,
+	StoreCallback,
+	ValidatorsAPI,
+} from '../../types';
+import {
+	computeValidatorsHash,
+	getIDAsKeyForStore,
+	rawStateStoreKey,
+	sortValidatorsByBLSKey,
+	updateActiveValidators,
+	validateFormat,
+} from '../../utils';
+import { SidechainInteroperabilityStore } from '../store';
 
-export class MainchainCCUpdateCommand extends BaseCommand {
+export class SidechainCCUpdateCommand extends BaseInteroperabilityCommand {
 	public name = 'sidechainCCUpdate';
 	public id = COMMAND_ID_SIDECHAIN_CCU;
 	public schema = crossChainUpdateTransactionParams;
+	private _bftAPI!: BFTAPI;
+	private _validatorsAPI!: ValidatorsAPI;
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async execute(_context: CommandExecuteContext<unknown>): Promise<void> {
-		throw new Error('Method not implemented.');
+	public addDependencies(args: CrossChainCommandDependencies) {
+		this._bftAPI = args.bftAPI;
+		this._validatorsAPI = args.validatorsAPI;
+	}
+	public async verify(
```

Comment:
I think we can unify the verify code somehow with mainchain verify since the logic is exactly the same

Reply:
I'll try to do it in Store class or as a utill func